File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp; 
        int n = nums.size() / 2; 

        
        for (auto a : nums) {
            mp[a]++;
        }

        int ans = 0; 

        
        for (auto a : mp) {
            if (a.second > n) {
                ans = a.first; 
            }
        }

        return ans; 
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        int vib = nums.length;
        for (int i = 0; i<nums.length; i++) {
            int num = nums[i];
            hm.put(num, hm.getOrDefault(num, 0) + 1);
        }

        for (int i = 0; i<nums.length; i++) {
            int num = nums[i];
            if (hm.get(num) > vib/2) {
                return num;
            }
        }
        return -1;        
    }
}